pages/search.js

```diff
@@ -150,10 +134,34 @@ export default function Search({
       );
       fetchUsers(inputValue);
     }, 500);
-
-    return () => clearTimeout(timer);
+    console.log(inputValue);
+    return () => {
+      clearTimeout(timer);
+    };
   }, [inputValue]);
+  async function fetchUsers(value) {
+    try {
+      setLoading(true);
+      const res = await fetch(
+        `${BASE_URL}/api/search?${new URLSearchParams({
+          slug: value,
+        }).toString()}`,
+      );
+      const data = await res.json();
+      if (data.error) {
+        throw new Error(`${value} not found`);
+      }
 
+      setNotFound();
+      setUsers(data.users.sort(() => Math.random() - 0.5));
+      setCurrentPage(1);
+    } catch (err) {
+      setNotFound(err.message);
+      setUsers([]);
+    } finally {
+      setLoading(false);
+    }
+  }
```

Comment:
I'm not sure why you moved this function outside from useEffect?

Reply:
Beacuse we are using fetchuser only in timer,to make code cleaner,it doesnt effect any function